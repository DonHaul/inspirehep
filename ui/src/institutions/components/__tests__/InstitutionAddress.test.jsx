import React from 'react';
import { render } from '@testing-library/react';
import { fromJS } from 'immutable';
import InstitutionAddress from '../InstitutionAddress';

describe('InstitutionAddress', () => {
  it('renders with only postal address', () => {
    const address = fromJS({
      postal_address: ['Rue Einstein', 'CH-1211 Genève 23'],
    });
    const { asFragment } = render(<InstitutionAddress address={address} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with only country', () => {
    const address = fromJS({
      country: 'Switzerland',
    });
    const { asFragment } = render(<InstitutionAddress address={address} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with country already in postal address', () => {
    const address = fromJS({
      postal_address: ['Rue Einstein', '123 SWITZERLAND'],
      country: 'Switzerland',
    });
    const { asFragment } = render(<InstitutionAddress address={address} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with city already in postal address', () => {
    const address = fromJS({
      postal_address: ['Rue Einstein', '123 meyrin'],
      cities: ['Meyrin'],
    });
    const { asFragment } = render(<InstitutionAddress address={address} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders all', () => {
    const address = fromJS({
      postal_address: ['Rue Einstein', 'CH-1211 Genève 23'],
      cities: ['Meyrin'],
      country: 'Switzerland',
      state: 'Geneva',
      place_name: 'CERN',
    });
    const { asFragment } = render(<InstitutionAddress address={address} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders empty', () => {
    const address = fromJS({});
    const { asFragment } = render(<InstitutionAddress address={address} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
