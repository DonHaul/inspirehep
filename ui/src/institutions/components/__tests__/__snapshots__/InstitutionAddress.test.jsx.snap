// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstitutionAddress renders all 1`] = `
<DocumentFragment>
  <span>
    <div
      class="__InlineList__ di"
    >
      <ul>
        <li>
          <div
            class="__InlineList__"
            data-testid="inline-data-list"
          >
            <ul>
              <li>
                Rue Einstein, 
              </li>
              <li>
                CH-1211 Genève 23
              </li>
            </ul>
          </div>
          , 
        </li>
        <li>
          <span>
            Meyrin
          </span>
          , 
        </li>
        <li>
          <span>
            Switzerland
          </span>
        </li>
      </ul>
    </div>
  </span>
</DocumentFragment>
`;

exports[`InstitutionAddress renders empty 1`] = `
<DocumentFragment>
  <span>
    <div
      class="__InlineList__ di"
    >
      <ul />
    </div>
  </span>
</DocumentFragment>
`;

exports[`InstitutionAddress renders with city already in postal address 1`] = `
<DocumentFragment>
  <span>
    <div
      class="__InlineList__ di"
    >
      <ul>
        <li>
          <div
            class="__InlineList__"
            data-testid="inline-data-list"
          >
            <ul>
              <li>
                Rue Einstein, 
              </li>
              <li>
                123 meyrin
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </span>
</DocumentFragment>
`;

exports[`InstitutionAddress renders with country already in postal address 1`] = `
<DocumentFragment>
  <span>
    <div
      class="__InlineList__ di"
    >
      <ul>
        <li>
          <div
            class="__InlineList__"
            data-testid="inline-data-list"
          >
            <ul>
              <li>
                Rue Einstein, 
              </li>
              <li>
                123 SWITZERLAND
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </span>
</DocumentFragment>
`;

exports[`InstitutionAddress renders with only country 1`] = `
<DocumentFragment>
  <span>
    <div
      class="__InlineList__ di"
    >
      <ul>
        <li>
          <span>
            Switzerland
          </span>
        </li>
      </ul>
    </div>
  </span>
</DocumentFragment>
`;

exports[`InstitutionAddress renders with only postal address 1`] = `
<DocumentFragment>
  <span>
    <div
      class="__InlineList__ di"
    >
      <ul>
        <li>
          <div
            class="__InlineList__"
            data-testid="inline-data-list"
          >
            <ul>
              <li>
                Rue Einstein, 
              </li>
              <li>
                CH-1211 Genève 23
              </li>
            </ul>
          </div>
        </li>
      </ul>
    </div>
  </span>
</DocumentFragment>
`;
